lib/chef/knife/ssh.rb

```diff
@@ -247,18 +247,18 @@ def search_nodes
       # @since 12.5.0
       # @param host [String] Hostname for this session.
       # @param port [String] SSH port for this session.
-      # @param type [Integer] Flag: host or gateway key
       # @param user [String] Optional username for this session.
+      # @param gatway [Boolean] Flag: host or gateway key
       # @return [Hash<Symbol, Object>]
-      def session_options(host, port, type = 0, user = nil)
+      def session_options(host, port, user = nil, gateway = false)
         ssh_config = Net::SSH.configuration_for(host, true)
         {}.tap do |opts|
           # Chef::Config[:knife][:ssh_user] is parsed in #configure_user and written to config[:ssh_user]
           opts[:user] = user || config[:ssh_user] || ssh_config[:user]
-          if type == 0 && config[:ssh_identity_file]
+          if gateway == false && config[:ssh_identity_file]
```

Comment:
`if !gateway && config[:ssh_identity_file]` is more idiomatic ruby in this case.